packages/cozy-pouch-link/src/CozyPouchLink.js

```diff
@@ -556,18 +556,28 @@ class PouchLink extends CozyLink {
       res = withoutDesignDocuments(res)
       withRows = true
     } else {
+      let findSelector = selector
+      const shouldAddId = !findSelector
+      if (shouldAddId) {
+        findSelector = {}
+      }
       if (indexedFields) {
         for (const indexedField of indexedFields) {
-          if (!Object.keys(selector).includes(indexedField)) {
-            selector[indexedField] = {
+          if (!Object.keys(findSelector).includes(indexedField)) {
+            findSelector[indexedField] = {
               $gt: null
             }
           }
         }
       }
+      if (shouldAddId) {
+        findSelector['_id'] = {
+          $gt: null
+        }
+      }
```

Comment:
You don't need to force the `_id` selector if the mechanism on line 566-567 added a predicate.
Thus, we might refacto like this:
```js
let findSelector = selector || {}
// ...do the stuff on lines 566...
findSelector = Object.keys(selector).length > 0 ? findSelector : { _id: { $gt: null } } // PouchDB does not accept empty selector 
```

And this whole behavior should definitely be tested, as suggested in my previous comment :wink:

Reply:
Done in #1524